src/roulette.c

```diff
@@ -2236,386 +1818,77 @@ void sub_8116EF8(u8 r0)
         for (i = (r0 + 0x1); i < (r0 + 0x5); i++)
             if (!(eRoulette->var08 & gUnknown_083F8C00[i].var08))
                 var0 |= gUnknown_083F8C00[i].var10;
-        var0 &= 0xDFFF;
-        sub_8124CE8(&eRoulette->varB8, var0);
+        sub_8124CE8(&eRoulette->varB8, var0 &= 0xDFFF);
         break;
     default:
-        memcpy(var1, gUnknown_083F8E9C, 0x18);
-        if ((u8)(r0 - 0x1) < 0x4)
-            var2 = 0x3;
+    {
+        struct UnkStruct1 var1[0x3];
+        memcpy(var1, gUnknown_083F8E9C, sizeof var1);
+        if (r0 > 0 && r0 < 5)
+            var2 = 3;
         else
-            var2 = 0x1;
-        var3 = ((r0 / 0x5) - 0x1);
-        switch((u8)r0 % 0x5)
+            var2 = 1;
+        var3 = r0 / 5 - 1;
+        switch (r0 % 5)
         {
-        case 0x1:
-            var3 = gSprites[eRoulette->var3C[0x7 + 0x0]].oam.paletteNum * 0x10;
-            break;
-        case 0x2:
-            var3 = gSprites[eRoulette->var3C[0x7 + 0x1]].oam.paletteNum * 0x10;
-            break;
-        case 0x3:
-            var3 = gSprites[eRoulette->var3C[0x7 + 0x2]].oam.paletteNum * 0x10;
-            break;
-        case 0x4:
-            var3 = gSprites[eRoulette->var3C[0x7 + 0x3]].oam.paletteNum * 0x10;
-            break;
+            case 0x1:
```

Comment:
extra spaces

Reply:
you know my position on indentation within switch contexts; i stand by it